Sources/Testing/Testing.docc/MigratingFromXCTest.md

```diff
@@ -127,9 +127,13 @@ family of functions. When writing tests using the testing library, implement
   }
 }
 
-The use of `async` and `throws` is optional. If teardown is needed, declare your
-test suite as a `final` class or as an actor rather than as a structure and
-implement `deinit`:
+In XCTest, you can also schedule work to run after a test using the [`tearDown()`](https://developer.apple.com/documentation/xctest/xctest/3856482-teardown)
+family of functions. When writing tests using the testing library, adopt structured
+  [concurrency](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/concurrency/) 
+and take advantage of Swift to handle cleanup for you where possible rather 
```

Comment:
Can you clarify what feature(s) of Swift concurrency, or which APIs, you mean here which assist with cleanup? I can think of `defer`, but I would consider that a basic language feature rather than a _concurrency_ feature.

Reply:
I think what we should aim to push people towards is using `with` style methods e.g. `try await withNetworkConnection { connection in ... }`. This is the pattern that recommend to people to have guaranteed lifetimes with structured concurrency.

Reply:
That's something they have to write themselves, and explaining how to write one would get us into the weeds a bit.

Reply:
There is precedent for describing how testing helps you design nicer APIs and vice versa, and it's something Apple has already published in the Xcode documentation: https://developer.apple.com/documentation/xcode/updating-your-existing-codebase-to-accommodate-unit-tests

So maybe we could take a similar approach here, where we show a `with…` function as an example of designing API+test together, but (as a separate task) create another doc that shows the thinking behind that design?

Reply:
Would this be an appropriate place to use a [snippet](https://github.com/swiftlang/swift-evolution/blob/main/proposals/0356-swift-snippets.md)?

Reply:
I think an example here would be extremely helpful for users trying to migrate